### DISCUSSION TITLE
Add option to confine cursor to screen area when letterboxed

### DISCUSSION THREAD

**Discussion by Stedoss** [ppy/osu#16955] [Ideas] [answered]
I play the game at 4:3, and noticed that there doesn't seem to be an option to confine the cursor to the screen area when letterboxed; it will confine to the window itself, meaning the user is able to "click" on the black screen area outside of the rendered game.

In stable, when `Render at native resolution` is checked, and the game is in fullscreen, the cursor will confine to the game area. Unsure whether this should be the default behaviour here, but there should at least be an option for it imo.

It's not really a huge deal, but it's a little strange that the cursor dissapears into the borders and can still seem to interact with the game still.

**Comment by peppy** [comment, marked as answer]
When you say "letterboxed", do you mean when applying "everything" screen scaling?

Can you provide a screenshot of your osu! window?

**Reply by Stedoss** [reply]
Sorry, I should have included some more info on that. Yes, this is when the "everything" screen scaling setting is applied.

![osu_2022-02-22_04-19-10](https://user-images.githubusercontent.com/29103029/155062526-7f13ce9d-3f72-4c65-9ccc-782377075c89.png)

**Reply by peppy** [reply]
I think this is a valid assumption. Probably should only be the case with "Everything" scaling, and not gameplay/exclude-overlays.

Going to require using clip rectangle specifications, which is likely going to introduce a lot more issues when it comes to relative mouse handling, but eventually required.

Tracking via https://github.com/ppy/osu/issues/16956.
